From 71c647d541e94ddd6d07a4c510ec743d249defbc Mon Sep 17 00:00:00 2001
From: sjorge <sjorge@blackdot.be>
Date: Fri, 27 Sep 2019 15:45:13 +0200
Subject: [PATCH] OS-XXXX Do not destroy external delegated datasets

The dataset created by the `delegate_dataset` property lives inside
 the zonepath so it will be destroyed by cuninstall.

For the other datasets instead of destroying them we inherit the
 zoned property so they show up again in the global zone.
---
 usr/src/lib/brand/jcommon/libhooks.ksh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/usr/src/lib/brand/jcommon/libhooks.ksh b/usr/src/lib/brand/jcommon/libhooks.ksh
index ec49fb9e3d..add9827a66 100644
--- a/usr/src/lib/brand/jcommon/libhooks.ksh
+++ b/usr/src/lib/brand/jcommon/libhooks.ksh
@@ -87,10 +87,10 @@ jattach_zone_final_setup()
 
 function juninstall_delegated_dataset
 {
-	# Now destroy any delegated datasets. Redirect to /dev/null in case they
-	# were already destroyed when we removed the zonepath dataset.
+	# Reset the zoned property for any delegated datasets. Redirect to /dev/null in case they
+	# any were destroyed when we removed the zonepath dataset.
 	DD=`zonecfg -z $ZONENAME info dataset | nawk '{if ($1 == "name:") print $2}'`
 	for i in $DD; do
-		zfs destroy -rF $i >/dev/null 2>&1
+		zfs inherit zoned -r $i >/dev/null 2>&1
 	done
 }
-- 
2.21.0

